warehouse/router/upload_test.go

```diff
@@ -127,17 +126,30 @@ func TestColumnCountStat(t *testing.T) {
 		tc := tc
 
 		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
 			conf := config.New()
 			conf.Set(fmt.Sprintf("Warehouse.%s.columnCountLimit", strings.ToLower(warehouseutils.WHDestNameMap[tc.destinationType])), tc.columnCountLimit)
 
-			j := UploadJob{
-				conf: conf,
-				upload: model.Upload{
+			pool, err := dockertest.NewPool("")
+			require.NoError(t, err)
+
+			pgResource, err := postgres.Setup(pool, t)
+			require.NoError(t, err)
+
+			uploadJobFactory := &UploadJobFactory{
+				logger:       logger.NOP,
+				statsFactory: statsStore,
+				conf:         conf,
+				db:           sqlmiddleware.New(pgResource.DB),
+			}
+			rs := redshift.New(config.New(), logger.NOP, stats.NOP)
```

Comment:
Can we use a manager in here?
```suggestion
			whManager, err := manager.New(warehouseutils.POSTGRES, conf, logger.NOP, statsStore)
			require.NoError(t, err)
```

Reply:
Replaced redshift with manager. I had used redshift because that is how other [tests](https://github.com/rudderlabs/rudder-server/pull/5426/files#diff-3ca4ffe249a74c6545b23670aa07940de31b9f23362020de94830c8f10c7aafeR258) were written

Reply:
Yeah, we would need to fix other tests as well.